runtime/compiler/z/codegen/J9CodeGenerator.cpp

```diff
@@ -184,7 +184,7 @@ J9::Z::CodeGenerator::CodeGenerator() :
 
    const bool accessStaticsIndirectly = !comp->target().cpu.isAtLeast(OMR_PROCESSOR_S390_Z10) ||
          comp->getOption(TR_DisableDirectStaticAccessOnZ) ||
-         (comp->compileRelocatableCode() && !comp->getOption(TR_UseSymbolValidationManager));
+         ((comp->compileRelocatableCode() || comp->isOutOfProcessCompilation()) && !comp->getOption(TR_UseSymbolValidationManager));
```

Comment:
`accessStaticsIndirectly `'s behaviour is consisted across all codgens except for Z. It is not enabled on the other platforms by default. When I try to disable it on Z by removing the `comp->isOutOfProcessCompilation()` here, I see crashes in tests. Currently investigating the failures.

Reply:
You are tying `outOfProcessCompilation` to `SVM`. Maybe the correct solution is `(comp->compileRelocatableCode() && !comp->getOption(TR_UseSymbolValidationManager) ||  comp->isOutOfProcessCompilation()`, though if the code works ok for AOT with SVM it should work for JITServer as well.

Reply:
Yes. Apologies! This was just a placeholder while I was trying to narrow the problem down.

Reply:
When enabling access to statics via the property set here, I encountered a problem when looking at trees like the following:
```
n18n     (  0)  istore  <recompilation-counter>[#305  Static] [flags 0x1000303 0x40 ]                [     0x3ff7ad029b0] bci=[-1,0,33] rc=0 vc=115 vn=- li=4 udi=- nc=1
 n17n     (  2)    iadd                                                                               [     0x3ff7ad02960] bci=[-1,0,33] rc=2 vc=115 vn=- li=4 udi=- nc=2
 n15n     (  1)      iload  <recompilation-counter>[#305  Static] [flags 0x1000303 0x40 ] (cannotOverflow )  [     0x3ff7ad028c0] bci=[-1,0,33] rc=1 vc=115 vn=- li=4 udi=- nc=0 flg=0x1000
 n16n     (  1)      iconst -1 (X!=0 X<=0 )                                                           [     0x3ff7ad02910] bci=[-1,0,33] rc=1 vc=115 vn=- li=4 udi=- nc=0 flg=0x204
------------------------------
------------------------------
 n18n     (  0)  istore  <recompilation-counter>[#305  Static] [flags 0x1000303 0x40 ]                [     0x3ff7ad029b0] bci=[-1,0,33] rc=0 vc=115 vn=- li=4 udi=- nc=1
 n17n     (  1)    iadd (in GPR_0160)                                                                 [     0x3ff7ad02960] bci=[-1,0,33] rc=1 vc=115 vn=- li=4 udi=- nc=2
 n15n     (  0)      iload  <recompilation-counter>[#305  Static] [flags 0x1000303 0x40 ] (in GPR_0160) (cannotOverflow )  [     0x3ff7ad028c0] bci=[-1,0,33] rc=0 vc=115 vn=- li=4 udi=- nc=0 flg=0x1000
 n16n     (  0)      iconst -1 (X!=0 X<=0 )                                                           [     0x3ff7ad02910] bci=[-1,0,33] rc=0 vc=115 vn=- li=4 udi=- nc=0 flg=0x204
------------------------------
 [     0x3ff7ae40e90]   LRL     GPR_0160,0x0000000000000000
 [     0x3ff7ae40fa0]   AHI     GPR_0160,0xffff
 [     0x3ff7ae41090]   STRL    GPR_0160,0x0000000000000000
```

Trees like this are created in fall back path here:
https://github.com/eclipse/omr/blob/24cd6ede529b51b68f8af33474c7006525cb8c86/compiler/il/OMRTreeTop.cpp#L90-L96
And here:
https://github.com/eclipse/omr/blob/24cd6ede529b51b68f8af33474c7006525cb8c86/compiler/il/OMRTreeTop.cpp#L117-L122

When evaluating these trees, we end up in here inside the `iloadHelper` or `istoreHelper` routines:
https://github.com/eclipse/omr/blob/24cd6ede529b51b68f8af33474c7006525cb8c86/compiler/z/codegen/OMRTreeEvaluator.cpp#L4841-L4848
https://github.com/eclipse/omr/blob/24cd6ede529b51b68f8af33474c7006525cb8c86/compiler/z/codegen/OMRTreeEvaluator.cpp#L5563-L5566

Inside the `relativeLongLoadHelper` or relativeLongStoreHelper` routine, we will fall through a block of code like this:
https://github.com/eclipse/omr/blob/24cd6ede529b51b68f8af33474c7006525cb8c86/compiler/z/codegen/OMRTreeEvaluator.cpp#L4776-L4789

This is obviously incorrect because we will end up generating position dependent code (i.e. `LRL`). So we must not generate this instruction. The solution to this problem can be found in https://github.com/eclipse/omr/pull/4458.

Reply:
This check has now been removed.